### PR TITLE
version 7.5.0-SNAPSHOT

### PR DESCRIPTION
**Pull request by djuarezgf**
**What's in the PR**
* Version 7.5.0-SNAPSHOT

### CODE REVIEW

**Review by TasDeniz — APPROVED** [review]

File: src/main/java/de/samply/share/client/util/connector/CentralSearchConnector.java [hunk 1]
```diff
@@ -256,14 +260,38 @@ public DateRestriction getDateRestriction() throws CentralSearchConnectorExcepti
     }
   }
 
+  private UploadStats deserialize(String stringUploadStats)
+      throws IOException, ClassNotFoundException {
+
+    byte[] byteArrayUploadStats = Base64.getDecoder().decode(stringUploadStats);
+    InputStream inputStream = new ByteArrayInputStream(byteArrayUploadStats);
+    ObjectInputStream objectInputStream = new ObjectInputStream(inputStream);
+    UploadStats uploadStats = (UploadStats) objectInputStream.readObject();
```
Comment (alexanderkiel) [inline src/main/java/de/samply/share/client/util/connector/CentralSearchConnector.java:269]: This is Java Serialisation and not XML. Please use Jackson here.

File: src/main/java/de/samply/share/client/util/connector/CentralSearchConnector.java [hunk 2]
```diff
@@ -295,10 +323,8 @@ public void setLastUploadTimestamp(String timestamp) throws CentralSearchConnect
           SamplyShareUtils.addTrailingSlash(centralSearchUrl.getPath()) + PATH_UPLOAD_STATS);
 
       UploadStats uploadStats = new UploadStats(timestamp);
-      Serializer serializer = new Persister();
-      StringWriter writer = new StringWriter();
-      serializer.write(uploadStats, writer);
-      HttpEntity httpEntity = new StringEntity(writer.getBuffer().toString());
+      String stringUploadStats = serialize(uploadStats);
```
Comment (alexanderkiel) [inline src/main/java/de/samply/share/client/util/connector/CentralSearchConnector.java:326]: I would not introduce a Variable for the serialized upload stats here. Just put the serialize call into the StringEntity constructor call.

File: src/main/java/de/samply/share/client/model/centralsearch/UploadStats.java [hunk 3]
```diff
@@ -1,20 +1,15 @@
...
 
 /**
  * Represents the upload stats as received from central search.
  */
-@Root(name = "UploadStats")
-public class UploadStats {
 
-  @Element(name = "LastUploadTimestamp")
-  private String lastUploadTimestamp;
+public class UploadStats implements Serializable {
```
Comment (alexanderkiel) [inline src/main/java/de/samply/share/client/model/centralsearch/UploadStats.java:9]: Please do not use Java serialization. See below.

**Review by alexanderkiel — CHANGES_REQUESTED** [review]

**Comment by alexanderkiel** [comment]
Also: please name the PR according the things you did in it.

File: src/main/java/de/samply/share/client/util/connector/CentralSearchConnector.java (hunk 1 above)
Comment (djuarezgf) [reply src/main/java/de/samply/share/client/util/connector/CentralSearchConnector.java]: Implemented

**Comment by djuarezgf** [comment]
Originally, this PR was only intended to change the version of the "develop" branch. Before merging, we have tried to resolve open comments on previous closed PR. Therefore, the problem with the name. Probably, we should have merged this PR and created a new one for the other changes.

**Comment by alexanderkiel** [comment]
So why not create a new PR for the removal of the `simple-xml-safe` dependency? That would be very clear.

**Pull request closed without merging**